app/models/grda_warehouse/hud/client.rb

```diff
@@ -1778,17 +1778,23 @@ def total_days_of_service
       ((date_of_last_service - date_of_first_service).to_i + 1) rescue 'unknown' # rubocop:disable Style/RescueModifier
     end
 
-    def self.text_search(text, client_scope:)
-      text_searcher(text) do |where|
-        client_scope.
-          joins(:warehouse_client_source).searchable.
-          where(where).
-          preload(:destination_client).
-          map { |m| m.destination_client&.id }.
-          compact
-      rescue RangeError
-        return none
-      end
+    # @param client_scope [GrdaWarehouse::Hud::Client.source] source clients to search in
+    # @param sorted [Boolean] order results by closest match to text
+    def self.text_search(text, client_scope: nil, sorted: true)
```

Comment:
Why is `client_scope` allowed to be `nil` now? I'd worry that this could lead to someone doing a search that is not scoped down by access, which it always should be.

Reply:
Nevermind.. I think thats ok, thats how lots of scopes work